src/web/app/pages-instructor/instructor-student-list-page/instructor-student-list-page.component.ts

```diff
@@ -67,461 +27,115 @@ interface GetCourseDetailsResponse {
-  }
-
-  /**
-   * Function to get initial data of the instructor's courses.
+   * Loads courses of current instructor.
    */
-  loadCourses(user: string): void {
-    const paramMap: { [key: string]: string } = { user };
-    this.httpRequestService.get('/instructor/students/courses', paramMap)
-        .subscribe((resp: GetCourseResponse) => {
-          this.courses = resp.courses;
-
-          if (!this.courses) {
-            this.statusMessageService.showWarningMessage('You do not have any courses yet.');
-          }
+  loadCourses(): void {
+    this.courseService.getCoursesForInstructor('instructor', 'active')
+        .subscribe((courses: Courses) => {
+          this.courseList = courses;
+          this.viewStudent = Array<boolean>(this.courseList.courses.length).fill(false);
 
         }, (resp: ErrorMessageOutput) => {
           this.statusMessageService.showErrorMessage(resp.error.message);
         });
   }
 
   /**
-   * Change value of the checkbox Display Archive Courses.
-   */
-  toggleDisplayArchive(): void {
-    this.isDisplayArchive = !this.isDisplayArchive;
-    const isDisplayArchiveChecked: boolean = this.isDisplayArchive;
-    this.courses.forEach((course: Course) => {
-      if (course && course.isArchived) {
-        this.toggleCourseStateAtInput(course, isDisplayArchiveChecked);
-      }
-    });
-  }
-
-  /**
-   * Change states of all courses.
+   * Toggles specific card and loads students if needed
    */
-  toggleAllPresentCoursesStateAtInput(defaultState: boolean): void {
-    this.allPresentCourses.forEach((course: Course) => {
-      if (course) {
-        this.toggleCourseStateAtInput(course, defaultState);
-      }
-    });
-  }
-
-  /**
-   * Change the state of all sections presented on the screen.
-   */
-  toggleAllPresentSectionDetailsState(defaultState: boolean): void {
-    this.allPresentSectionDetails.forEach((sectionDetails: SectionDetails) => {
-      this.toggleSectionState(sectionDetails, defaultState);
-    });
-  }
-
-  /**
-   * Change the state of all teams presented on the screen.
-   */
-  toggleAllPresentTeamDetailsState(defaultState: boolean): void {
-    this.allPresentTeamDetails.forEach((teamDetails: TeamDetails) => {
-      this.toggleTeamState(teamDetails, defaultState);
-    });
-  }
-
-  /**
-   * Change the state of all students presented on the screen.
-   */
-  toggleAllPresentStudentDetailsState(defaultState: boolean): void {
-    this.allPresentStudentDetails.forEach((studentDetails: StudentDetails) => {
-      this.toggleStudentState(studentDetails, defaultState);
-    });
-  }
-
-  /**
-   * Trigger getting CourseDetails data for a course from backend.
-   */
-  toggleCourseStateAtInput(course: Course, designatedState?: boolean): void {
-    if (course.isChecked === undefined) {
-      course.isChecked = false;
+  toggleCard(index: number, courseId: string): void {
```

Comment:
Now, you can get rid of the index and mapping and change the signature to
```suggestion
  toggleCard(courseTab: CourseTab): void {
```
and manipulate the data inside the tab directly.

Reply:
Done!